mobilevit/models/mobilevit.py

```diff
@@ -0,0 +1,61 @@
+from black import out
+import tensorflow as tf
+from configs.model_config import get_model_config
+from tensorflow import keras
+
+from mobilevit.models.conv_block import conv_block, inverted_residual_block
+
+
+def get_training_model(
+    model_name: str,
+    image_shape: tuple,
+    num_classes: int,
+) -> keras.Model:
+    """
+    Implements MobileViT family of models given a configuration.
+    References:
+        (1) https://arxiv.org/pdf/2110.02178.pdf
+
+    Args:
+        model_name (str): Name of the MobileViT model
+        image_shape (tuple): Shape of the input image
+        num_classes (int): number of classes of the classifier
+
+    Returns:
+        model: Keras Model
+    """
+    configs = get_model_config(model_name)
+
+    input_layer = keras.Input(shape=image_shape)  # Input Layer
+
+    # Convolutional Stem Stage
+    x = conv_block(input_layer=input_layer, num_filters=configs.out_channels[0])
+
+    x = inverted_residual_block(
+        input_layer=x,
+        expanded_channels=configs.out_channels[0] * configs.expansion_factor,
+        output_channels=configs.out_channels[1],
+    )
+
+    # Downsample with Inverted Residual Block
+    x = inverted_residual_block(
+        x,
+        expanded_channels=configs.out_channels[1] * configs.expansion_factor,
+        output_channels=configs.out_channels[2],
+        strides=2,
+    )
+
+    x = inverted_residual_block(
+        x,
+        expanded_channels=configs.out_channels[2] * configs.expansion_factor,
+        output_channels=configs.out_channels[2],
+    )
+
+    x = inverted_residual_block(
+        x,
+        expanded_channels=configs.out_channels[2] * configs.expansion_factor,
+        output_channels=configs.out_channels[3],
+    )
+
+    model = keras.Model(input_layer, x)
```

Comment:
If I remember correctly, there's an interpolation step in the model if the resolution is 160 or 192. Be sure to take care of that.

Reply:
I checked with all the model sizes, like `XXS, XS, S`, the model is giving the desired output as reported in the paper.

Reply:
What about input resolution? The paper also mentions an interpolation step and it is reflected in the [official code](https://github.com/apple/ml-cvnets/blob/d38a116fe134a8cd5db18670764fdaafd39a5d4f/cvnets/modules/mobilevit_block.py#L125) too.

Reply:
To handle the interpolation, Can we use [this](https://www.tensorflow.org/addons/api_docs/python/tfa/image/interpolate_bilinear)?

Reply:
We can use `tf.image.resize()` as well or alternatively `tf.keras.layers.Resizing()`. If they don't serve the purpose, we can go with the tfa later.

Reply:
Make sense.

Reply:
I think this [line of code](https://github.com/apple/ml-cvnets/blob/d38a116fe134a8cd5db18670764fdaafd39a5d4f/cvnets/modules/mobilevit_block.py#L121) shows the new resolution to which the interpolation needs to happen?

Reply:
Interpolation happens in `MobileViT` block. It is not happening in these blocks. This PR is ready to go I guess.

Reply:
Feel free to merge then.